### PR TITLE
fix(curriculum): allow positive numbers only for decimal to binary conversion

### LINKED ISSUE #54651: refactor decimal to binary converter to check for positive numbers only
### Describe the Issue

***STEPS TO REPRODUCE:***
1. Add a negative number to your app 

***RESULT:***
Negative numbers are displayed as ```isNaN = false``` but the app displays NaN on the result field. 
The user can see on the console if the application running through the website of freeCodeCamp
```Potential infinite loop detected on line 32. Tests may fail if this is not changed.```

If I copy html, css and js file to vc and run it stand-alone on the browser I see this on the console
```
Uncaught RangeError: Maximum call stack size exceeded
    at decimalToBinary (script.js:33:3)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
    at decimalToBinary (script.js:36:12)
decimalToBinary @ script.js:33
decimalToBinary @ script.js:36
decimalToBinary @ script.js:36
decimalToBinary @ script.js:36
decimalToBinary @ script.js:36
...
```

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-recursion-by-building-a-decimal-to-binary-converter/step-109

### Your code

```JavaScript
const numberInput = document.getElementById("number-input");
const convertBtn = document.getElementById("convert-btn");
const result = document.getElementById("result");
const animationContainer = document.getElementById("animation-container");
const animationData = [
  {
    inputVal: 5,
    marginTop: 300,
    addElDelay: 1000,
    msg: 'decimalToBinary(5) returns "10" + 1 (5 % 2). Then it pops off the stack.',
    showMsgDelay: 15000,
    removeElDelay: 20000,
  },
  {
    inputVal: 2,
    marginTop: -200,
    addElDelay: 1500,
    msg: 'decimalToBinary(2) returns "1" + 0 (2 % 2) and gives that value to the stack below. Then it pops off the stack.',
    showMsgDelay: 10000,
    removeElDelay: 15000,
  },
  {
    inputVal: 1,
    marginTop: -200,
    addElDelay: 2000,
    msg: 'decimalToBinary(1) returns "1" (base case) and gives that value to the stack below. Then it pops off the stack.',
    showMsgDelay: 5000,
    removeElDelay: 10000,
  }
];

const decimalToBinary = (input) => {
  if (input === 0 || input === 1) {
    console.log("input ", input)
    console.log("input === 0 ", input === 0)
    console.log("input === 1 ", input === 1)

    return String(input);
  } else {
    const result = decimalToBinary(Math.floor(input / 2)) + (input % 2);
    console.log("result = >>>>>> ", result)
    return result
  }
};

const showAnimation = () => {
  result.innerText = "Call Stack Animation";

  animationData.forEach((obj) => {
    setTimeout(() => {
      animationContainer.innerHTML += `
        <p id="${obj.inputVal}" style="margin-top: ${obj.marginTop}px;" class="animation-frame">
          decimalToBinary(${obj.inputVal})
        </p>
      `;
    }, obj.addElDelay);

    setTimeout(() => {
      document.getElementById(obj.inputVal).textContent = obj.msg;
    }, obj.showMsgDelay);

    setTimeout(() => {
      document.getElementById(obj.inputVal).remove();
    }, obj.removeElDelay);
  });

  setTimeout(() => {

  }, 20000);
};

const checkUserInput = () => {
  const inputInt = parseInt(numberInput.value);
    console.log("inputInt=>>>>>", inputInt, " isNaN =>>>> ",isNaN(inputInt))
  if (!numberInput.value || isNaN(inputInt)) {
    alert("Please provide a decimal number");
    return;
  }

  if (inputInt === 5) {
    showAnimation();
    return;
  }

  result.textContent = decimalToBinary(inputInt);

  numberInput.value = "";
};

convertBtn.addEventListener("click", checkUserInput);

numberInput.addEventListener("keydown", (e) => {
  if (e.key === "Enter") {
    checkUserInput();
  }
});


```

### Expected behavior

As far as I read on Google negative numbers can be converted to binary but it's the whole process, 
Here is a clear example of how to do it at the bottom of the article: https://sciencing.com/calculate-binary-numbers-8267989.html

### Screenshots

***RESULT ON FREECODECAMP***
fromFreeCodeCamp

***RESULT WHEN COPIED TO LOCAL***
copiedToLocal

### System

Device: Laptop
OS: Windows 11 Home
Browser: Chrome
Version: 124.0.6367.61

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DeepaPrasanna**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54651

1. Added a note in Step 1 that conversion is for positive numbers only
image

2. Updated the ```if``` statement in required files
image

image

3. Updated the instructions in step 12
image

4. Updated the hint in step 12 when code does not pass
image
image

5. Success message when the code successfully passes
image

6. Updated hint in step 78
image
   
   on success, 
image

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/63db7f4677d06d7500a13321.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ dashedName: step-1
 
 In this project, you'll build a decimal and binary converter and learn about both number systems. You'll also learn about recursion by using it to perform the conversions.
 
+NOTE: This project will only convert positive numbers into binary.
+
 All of the HTML and CSS for this project has been provided for you.
 
 When you're ready to get started, use the `.getElementById()` method to get the input element with the id `number-input`, and store it in a variable called `numberInput`. Use the same method to get the button element with the id `convert-btn` and store it in a variable called `convertBtn`, and the `h2` element with the id `result` and store it in a variable called `result`.
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/63db7f4677d06d7500a13321.md:16]: Let's move the note down

```suggestion
In this project, you'll build a decimal and binary converter and learn about both number systems. You'll also learn about recursion by using it to perform the conversions.

All of the HTML and CSS for this project have been provided for you. When you're ready to get started, use the `.getElementById()` method to get the input element with the id `number-input`, and store it in a variable called `numberInput`. Use the same method to get the button element with the id `convert-btn` and store it in a variable called `convertBtn`, and the `h2` element with the id `result` and store it in a variable called `result`.

**NOTE**: This project will only convert positive numbers into binary.
```

**Comment by Ksound22** [comment]
This looks good. I'll approve when the conversation is resolved!

**Comment by DeepaPrasanna** [comment]
> This looks good. I'll approve when the conversation is resolved!

done!

**Review by Ksound22 — APPROVED** [review]

**Review by JoyShaheb — APPROVED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/64475c0b61cddb6feaab4e2e.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ dashedName: step-13
 
 Now you can alert the user if they don't enter a number, or the number is invalid before you attempt to convert it into binary.
 
-In the body of the `if` statement, use the `alert()` method to display the text `Please provide a decimal number`.
+In the body of the `if` statement, use the `alert()` method to display the text `please provide a decimal number greater than or equal to 0`.
```
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/64475c0b61cddb6feaab4e2e.md:12]:
```suggestion
In the body of the `if` statement, use the `alert()` method to display the text `Please provide a decimal number greater than or equal to 0`.
```

Can we bring the first word, first letter as capital please ? for all the steps

**Review by JoyShaheb — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6464a6d98c43acecb8a7867b.md [hunk 3]
```diff
@@ -208,8 +208,8 @@ const showAnimation = () => {
 const checkUserInput = () => {
   const inputInt = parseInt(numberInput.value);
 
-  if (!numberInput.value || isNaN(inputInt)) {
-    alert("Please provide a decimal number");
+  if (!numberInput.value || isNaN(inputInt) || inputInt < 0) {
+    alert("please provide a decimal number greater than or equal to 0");
     return;
```
Comment (JoyShaheb) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6464a6d98c43acecb8a7867b.md:213]:
```suggestion
```
it seems as if there's no utility for this `return` word. can you remove this and check if everything if properly functional or not ? 

Or, if i am missing something, then let me know

**Review by JoyShaheb — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/64475c0b61cddb6feaab4e2e.md (hunk 2 above)
Comment (DeepaPrasanna) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/64475c0b61cddb6feaab4e2e.md]: Done

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6464a6d98c43acecb8a7867b.md (hunk 3 above)
Comment (DeepaPrasanna) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6464a6d98c43acecb8a7867b.md:213]: As the return serves a purpose of stopping the function from continuing when the input is invalid, I am not removing it.

Comment (JoyShaheb) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-recursion-by-building-a-decimal-to-binary-converter/6464a6d98c43acecb8a7867b.md:213]: oh i see now ! my bad, i didn't see it coming. thank you @DeepaPrasanna

**Review by JoyShaheb — APPROVED** [review]

**Pull request merged**